### ISSUE TITLE
Frame send time * msec too slow, throttling maxfps

### ISSUE THREAD

**Issue by Jsalas424** [ZoneMinder/zoneminder#3225]
**Describe Your Environment**

- v1.34.26 
- Installed via PPA
- Ubuntu 18.04 in a Proxmox VM with 3 sockets x 2 cores and 3 GiB of RAM for just the 1 camera
- Firefox

**If the issue concerns a camera**

- Amcrest IP2M-841W configred with instructions from here: https://wiki.zoneminder.com/Amcrest#IP2M-841B.2FIP2M-841W
- 30fps
- 1920x1080
- ZoneMinder Source Type: RSTP
<img width="1037" alt="image" src="https://user-images.githubusercontent.com/51519895/116555091-f65b3f00-a8c9-11eb-96e3-4a9430c0398a.png">

<img width="782" alt="image" src="https://user-images.githubusercontent.com/51519895/116555623-9c0eae00-a8ca-11eb-8da9-7ba13a1148c5.png">

<img width="1420" alt="image" src="https://user-images.githubusercontent.com/51519895/116555231-20acfc80-a8ca-11eb-8ef9-d22321797918.png">

<img width="1424" alt="image" src="https://user-images.githubusercontent.com/51519895/116555288-33bfcc80-a8ca-11eb-916f-a1fe3d8fa7af.png">

**Describe the bug**

Logs are filled with error 'Frame send time * msec too slow, throttling maxfps' when viewing the stream via Firefox or ZMNinja

**To Reproduce**

**Expected behavior**

**Debug Logs**

Level 9 Debug logs for zms_m1: https://pastebin.com/uP63ByXe
<img width="1439" alt="image" src="https://user-images.githubusercontent.com/51519895/116555355-4934f680-a8ca-11eb-939f-97b1cf016564.png">

```

**Comment by Jsalas424** [comment]
Updated post with level 9 debug logs for zms_m1: https://pastebin.com/uP63ByXe

**Comment by connortechnology** [comment]
What is interesting here is that each frame gets progressively slower.  WHY?

A better approach might be to scale the image or otherwise reduce the size/complexity of the image.  This would save cpu as well as bandwidth.

**Comment by connortechnology** [comment]
@alabamatoy for the session.

This issue would have to be either cpu or bandwidth/latency.  We take the time, then encode the jpeg in ram, then print it out to the socket, take the time again.  If the amount of time is greater than the desired fps, we decrease the fps so that we wait less between sending frames.

Your graph shows pretty low cpu use and encoding jpeg's is not a time consuming operation. So I'm assuming network problems.

**Comment by Jsalas424** [comment]
I live in an apartment complex, 2.4GHz interference is definitely a reality and I'm assuming the wireless NIC in that Amcrest isn't the best, but there is only 30ft or so between the camera and the AP. I guess the only way to confirm is to hardwire it. I'll try to do that this weekend and report back.

**Comment by connortechnology** [comment]
as already stated, the issue is not with the camera. It is between your browser and ZM.  wifi is inherently unstable.  So wiring the camera in will improve your available wifi bandwidth but I suspect the issues will persist.

**Comment by Jsalas424** [comment]
> as already stated, the issue is not with the camera. It is between your browser and ZM. wifi is inherently unstable. So wiring the camera in will improve your available wifi bandwidth but I suspect the issues will persist.

I just reread that and saw the issue in my logic. So would a speed test between my end device (laptop on LAN wifi) and my ZM VM be provide more valuable information?

**Comment by Jsalas424** [comment]
Or am I being dense here and are you specifically referring to just the browser and not the connection between the end devices and the ZM VM? In that case I should have stated that when I generated these error logs, I was viewing my stream via ZMNinja not through my Firefox browser, although I can recreate this through Firefox if that is of value.

**Comment by Jsalas424** [comment]
Here I'm connecting from client to Zoneminder VM and getting 73 Mbits/s throughput tested with iperf3

<img width="568" alt="image" src="https://user-images.githubusercontent.com/51519895/117707396-8b004f80-b19c-11eb-83eb-29e7d11ed782.png">

**Comment by Jsalas424** [comment]
This is solved and was due a 2.4GHz band wifi interference issue

The Amcrest IP2M-841W only uses 2.4GHz making it prone to signal interference is crowded environment, for example my apartment building in a major city. 

I noticed that the network adapter for my ZM VM had increased usage right when I went on vacation:
<img width="568" alt="image" src="https://user-images.githubusercontent.com/51519895/121097825-9f972e00-c7c2-11eb-8829-c724980c5b4e.png">

and then tanked when I got back:
<img width="587" alt="image" src="https://user-images.githubusercontent.com/51519895/121097894-c35a7400-c7c2-11eb-8680-390ff8a41733.png">

My AP (Unifi Dream Machine) broadcasts 3 wifi channels as seperate VLANs. My main and guest channel broadcasts both on the 2.4GHz and 5GHz channels, but only at 2.4GHz for the camera channel. I turned off 2.4GHz on my main wifi channel since I know my main devices all support 5GHz anyways, and it fixed it!

<img width="575" alt="image" src="https://user-images.githubusercontent.com/51519895/121098248-70cd8780-c7c3-11eb-828f-9a875f40362b.png">
